templates/toil-instance.yaml

```diff
@@ -246,8 +247,76 @@ Resources:
               group: "root"
             /opt/toil/bin/reload-toil-cluster.sh:
               content: |
-                /opt/toil/bin/remove-toil-cluster.sh
-                /opt/toil/bin/create-toil-cluster.sh
+                /bin/bash /opt/toil/bin/remove-toil-cluster.sh
+                /bin/bash /opt/toil/bin/create-toil-cluster.sh
+              mode: "00644"
+              owner: "root"
+              group: "root"
+            /opt/toil/bin/add-security-group.py:
+              content: !Sub
+                - |
+                  #! /usr/bin/env python
+
+                  import boto3
+
+                  region = 'us-east-1'
+                  client = boto3.client('ec2', region_name=region)
+
+                  print('Ensure vpc security group is associated with leader node.')
+
+                  print('\nSending describe instances request')
+                  describe_instances_response = client.describe_instances(
+                    Filters=[
+                      {
+                          'Name': 'tag:Name',
+                          'Values': [
+                              "${ClusterName}",
+                          ]
+                      }
+                    ])
+
+                  print('Describe instances response:'')
+                  print(describe_instances_response)
+
+                  instance_ids = [
+                    instance.get('InstanceId')
+                    for reservation in describe_instances_response.get('Reservations')
+                    for instance in reservation.get('Instances') if instance.get('State').get('Name') != 'terminated'
+                  ]
+
+                  print('\nList of instance ids:')
+                  print(' '.join(instance_ids))
+                  assert len(instance_ids) == 1
+                  instance_id = instance_ids[0]
+                  print('\n\ninstance_id: %s' % instance_id)
+
+                  print('\n\Sending instance groups request')
+                  groups_response = client.describe_instance_attribute(
+                    InstanceId=instance_id,
```

Comment:
get all SG groups from a list of instance IDs. I'm not sure I understand this because different instances may contain different groups.  What would this list of groups give you?

Reply:
Because of the filter used above, the only things that will return to me are items in the cluster, which would all have the same SGs. I need to know what the existing SGs are in order to add on the one I want below. Because when you programmatically set SGs, you have to give the complete list, not just the one you want to append.

Reply:
ahh i see, all of them contain same SGs.  I guess i didn't understand why your getting SG from multiple nodes instead of just the leader node?  Or maybe I'm am just not understanding how this boto3 api works?

Reply:
Will not be an issue if I use ToilNodeType+Name above. I'll verify I have a length of 1 in the list.

Reply:
Also, if you look above I'm pulling the first instance id off anyway. And I'm asserting that the instance_ids list has a length of 1. 

This is the usual AWS api hijinks in my experience, boto really just wraps all that.